### ISSUE TITLE
Cumulative Sum Operation Limited to Rank-4 Tensors

### ISSUE THREAD

**Issue by karthikkattaopenxpu** [tenstorrent/tt-metal#22865]
### Describe the bug

The TTNN backend implementation of cumulative sum (aten.cumsum.default) has limitations that prevent it from handling most common tensor shapes and dimensions, causing test failures in PyTorch operation conformance tests.
Current Limitation
The implementation only supports:

Rank-4 tensors only (4-dimensional tensors)
Dimension 0 or 1 only for the cumulative sum operation

**Error Details**
Error Location: ox-src/examples/oxtosagpu/oxtosagpu/internal/ox_ttnn_backend/oxttnn_reduce_ops.cpp:57
Error Message: "Only rank 4 and dim == 0 or 1 are supported for CUM_SUM on TTNN."
Test Case: test_pytorch_op[cuda::bfloat16::aten.cumsum.default[2]]

**Impact**
This limitation breaks PyTorch compatibility for cumsum operations on:

Tensors with rank != 4 (1D, 2D, 3D, 5D+ tensors)
Cumsum along dimensions 2, 3, or higher (even for 4D tensors)
Common use cases like cumsum on vectors, matrices, or along height/width dimensions

Fails with runtime error for most tensor shapes and dimensions

Could you please let us know if there are plans to support other tensor ranks for the cumulative sum operation?

### Steps to reproduce the issue

```
import torch
tensor = torch.tensor([[1.0, 2.0, 3.0], 
                       [4.0, 5.0, 6.0]], dtype=torch.bfloat16).to("cuda")
result = torch.cumsum(tensor, dim=0)  # Cumsum along rows

```
### Expected behavior

PyTorch's torch.cumsum() supports:

Any tensor rank (1D, 2D, 3D, 4D, 5D+)
Any valid dimension for the tensor rank
All data types supported by the backend

**Comment by nmauriceTT** [comment]
Hello. Thank you for reaching out.

Yes, there are plans to add support for more tensor ranks & dim. 
Right now, there is [`ttnn.experimental.cumsum`](https://docs.tenstorrent.com/tt-metal/latest/ttnn/ttnn/api/ttnn.experimental.cumsum.html#ttnn.experimental.cumsum). It supports up to 5D tensors, and any `dim` parameter for float32/bfloat16 data types.

However, it is still in `experimental` and it is currently single-threaded. 
I've been working on a parallel version over the past few days, and intend to submit a patch for it this week.

**Comment by cmaryanTT** [comment]
@karthikkattaopenxpu - for our prioritization, is this gating your top priority models, or just an observation from your testing effort?

**Comment by karthikkattaopenxpu** [comment]
@cmaryanTT Just tested the experimental API—tests are passing successfully up to rank 4. I can't confirm yet whether any models require a rank higher than 5. So, consider this a preliminary test observation.

**Comment by cmaryanTT** [comment]
Marking this as P1 for implementation of rank < 4; Rank > 4 is P2.

**Comment by satya-openxpu** [comment]
While experimental cumsum is working, we are seeing accuracy issues when the output is compared against torch output. Here is the sample code that can replicate the mismatch (a max ULP difference of 5).

```
import torch
import ttnn

def calculate_ulp_max(val, ref):
    val, ref = val.cpu(), ref.cpu()
    val_i = val.to(dtype=torch.float32).view(dtype=torch.int32)
    ref_i = ref.to(dtype=torch.float32).view(dtype=torch.int32)
    if (val.dtype == torch.bfloat16) or (val.dtype == torch.float16):
        val_i = val_i >> 16
        ref_i = ref_i >> 16
        val_sign = (val_i & 0x8000)
        ref_sign = (ref_i & 0x8000)
        val_steps = (val_i & 0x7fff)
        ref_steps = (ref_i & 0x7fff)
    else:
        val_sign = (val_i & 0x80000000)
        ref_sign = (ref_i & 0x80000000)
        val_steps = (val_i & 0x7fffffff)
        ref_steps = (ref_i & 0x7fffffff)
    same_sign_mask = (val_sign == ref_sign)
    ulp = torch.where(same_sign_mask, torch.abs(val_steps-ref_steps), torch.abs(val_steps)+torch.abs(ref_steps)).to(dtype=torch.float32)
    ulp = torch.where(val.isnan(), float('nan'), ulp)
    ulp = torch.where(val.isinf(), float('nan'), ulp)
    ulp_max = torch.max(ulp).item()
    return ulp_max

device_id = 0
device = ttnn.open_device(device_id=device_id)

torch.manual_seed(404)
torch_tensor = torch.rand([1,59], dtype=torch.bfloat16, device="cpu")
outputs_torch = torch.ops.aten.cumsum.default(torch_tensor, dim=1)

ttnn_tensor = ttnn.from_torch(torch_tensor, layout=ttnn.TILE_LAYOUT, device=device)
ttnn_output = ttnn.experimental.cumsum(ttnn_tensor, dim=1)
ttnn_output = ttnn.to_layout(ttnn_output, ttnn.ROW_MAJOR_LAYOUT)

ttnn_output_torch = ttnn.to_torch(ttnn_output)

ulp_error = calculate_ulp_max(ttnn_output_torch, outputs_torch)
print(f"Max ULP difference: {ulp_error}")

ttnn.close_device(device)
```
